### PR TITLE
CF - integrate sass

### PR DESCRIPTION
**Pull request by c-franklin**
### Integrate sass

1. Installed sass

2. Created
- `scss/lib/functions`
- `scss/lib/vars` - colours, fonts, breakpoints
- `scss/lib/colors` - classes for colour and fill to style svgs

3. Refactored colours and fonts 
Colors that were hard-coded into the css -> now using sass functions.
`TODO` check all `.tsx` files for hard-coded colours
This can also be done to standardise font sizes. 

In the file `sass/lib/vars.scss` I have deliberately added in every single colour I found in the codebase and not tried to rationalise anything. See below. 

Ideally want to speak to your designer to create a palette and get colours into rgba(171,71,188 ,1) format.

Colour vars are named by element for theming purposes `_color(element-name)`
If you name something `graph-fill-yellow` it breaks when you want to change it from  `#FFD600` to `#8E24AA`. Instead you change the colour of `graph-fill-secondary` from  `#FFD600` to `#8E24AA`.

I have used the standard sass syntax of lowercase kebab-case for classes
https://sass-lang.com/documentation/style-rules

We could use pure `sass` vs `scss` if you prefer - that would involve a wider refactor (removing curly braces)

There's plenty more to do to reduce code duplication and create atomic libraries (hence the creation of the `scss` dir), but wanted to do an initial pass for feedback before going any further.

`All the colours I found`

Screen Shot 2021-02-22 at 12 52 02 AM

### CODE REVIEW

**Comment by c-franklin** [comment]
Checking linting now.

**Comment by delta1** [comment]
Thanks @charlottefranklin this looks really great! 

> We could use pure sass vs scss if you prefer 

Personally I prefer scss, so no need to go full sass 😆 

Will do a review and test today, and barring any issues should get it merged.

File: README.md [hunk 1]
```diff
@@ -26,3 +26,23 @@ You'll need a running [blockchain-explorer-api](https://github.com/tari-project/
...
+    
+    REACT_APP_EXPLORER_API_DOMAIN=explore.tari.com/api
+    REACT_APP_EXPLORER_API_PREFIX=https
+    REACT_APP_EXPLORER_WS_PREFIX=ws
+    REACT_APP_TOKEN_NAME=tXTR
+
+    # Run
+    
+    $ npm run build
+    $ serve -s build
```
Comment (delta1) [inline README.md]: Is this necessary? Does it not just work with `npm start`? I thought I tested it like that.. 

I don't think `serve` is a built-in unless the user has done `npm i -g serve`

**Review by delta1 — APPROVED** [review]
Thank you this is awesome

**Comment by delta1** [comment]
Hey @charlottefranklin - could you please squash these commits into 1 and [sign](https://docs.github.com/en/github/authenticating-to-github/about-commit-signature-verification) that commit? Signing is a requirement for our base branch (sorry I forgot about it). Let me know if you can do so, otherwise I can do it separately. Thanks again

**Comment by c-franklin** [comment]
> Hey @charlottefranklin - could you please squash these commits into 1 and [sign](https://docs.github.com/en/github/authenticating-to-github/about-commit-signature-verification) that commit? Signing is a requirement for our base branch (sorry I forgot about it). Let me know if you can do so, otherwise I can do it separately. Thanks again

Yep - will do now. Great.

File: README.md (hunk 1 above)
Comment (c-franklin) [reply README.md]: ooh need to check that - react told me to run serve -s build after I ran the build. Will take it out for now.

Comment (delta1) [reply README.md]: So afaik, `react-scripts start` (which is the start script from `package.json` in this case) starts a server and continuously builds the source on changes, with hot module reloading. So the `build` step is actually just for deployment.

**Comment by c-franklin** [comment]
Hey @delta1 I've squashed the commits and signed. 
Are you ok for me to force push this squashed commit?

**Comment by Krakaw** [comment]
> Hey @delta1 I've squashed the commits and signed.
> Are you ok for me to force push this squashed commit?

Yes, please.

**Review by Krakaw — APPROVED** [review]
This looks great, thanks! Just the comment from @delta1 that you've addressed.

**Comment by c-franklin** [comment]
checking conflicts now

**Comment by delta1** [comment]
Thanks @charlottefranklin! If you have any time to look at other [issues](/tari-project/block-explorer-frontend/issues) please go ahead

**Pull request merged**